### PR TITLE
Docker build

### PR DESCRIPTION
**Pull request by razor-x**
- Move Database type export
- Add docker support to codespace
- Add .dockerignore
- Add Docker workflows
- Update README for docker
- Test container with curl
- Add GitHub actions note to README
- Add Dockerfile
- Build server.ts
- Implement server.ts

Based on https://github.com/makenew/nodets-server

### CODE REVIEW

**Comment by razor-x** [comment]
I had to remove dpdm. Not sure why but it was trying to run the script. I'd like to understand but I removed it to unblock. I'm not sure we need dpdm check here though since we build the cjs to a single file.

**Comment by razor-x** [comment]
Build is a bit slow. In theory cache could speed things up. Alternative options

1. Only run build on main or tag. Downside is the build can break if it's not in each PR.
2. Reuse the package artifact that is built in the build step. Downside: docker build doesn't work without running the build step outside of docker, so only really works in ci.

File: .github/workflows/prune.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    steps:
+      - name: Prune smoke test images
+        uses: snok/container-retention-policy@v2
+        with:
+          image-names: smoke-*
+          cut-off: One week ago UTC
+          keep-at-least: 1
+          account-type: org
+          org-name: ${{ github.repository_owner }}
+          token: ${{ secrets.GH_TOKEN }}
```
Comment (seveibar) [inline .github/workflows/prune.yml:23]: hmm not sure we want this- it would run across all containers for our org? Feels like maybe not the right repo?

**Review by seveibar — APPROVED** [review]

File: .github/workflows/prune.yml (hunk 1 above)
Comment (razor-x) [reply .github/workflows/prune.yml:23]: Not sure I understand what you mean. Can you clarify?

**Comment by razor-x** [comment]
For the slowness, we can merge this now to get the container published, and experiment with speedup if we find it annoying. We don't have to require the new status check for now. It's likely to pass every time if the other ones do anyway.

File: .github/workflows/prune.yml (hunk 1 above)
Comment (razor-x) [reply .github/workflows/prune.yml:23]: You are right, it should have different options.

Comment (razor-x) [reply .github/workflows/prune.yml:24]: I think this should do it [`e78f1ef` (#38)](https://github.com/seamapi/fake-seam-connect/pull/38/commits/e78f1ef212a95fcde9c0aa2828ef364657767da6)

**Comment by codetheweb** [comment]
> Build is a [bit slow.](https://github.com/seamapi/fake-seam-connect/actions/runs/5675187248/attempts/1)

guessing it's slow mostly due to the arm64 build, would recommend using buildjet native arm64 runners

File: .github/actions/setup-docker/action.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      uses: docker/setup-qemu-action@v2
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+    - name: Setup cache
+      uses: actions/cache@v3
+      with:
+        path: /tmp/.buildx-cache
+        key: ${{ runner.os }}-buildx-${{ github.sha }}
+        restore-keys: |
+          ${{ runner.os }}-buildx-
```
Comment (codetheweb) [inline .github/actions/setup-docker/action.yml:29]: I think the build and push action now has native support for this type of caching

**Pull request merged**